miner/tx_pool_w_accounts.go

```diff
@@ -0,0 +1,111 @@
+package miner
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/pending_txs"
+	"github.com/spacemeshos/sha256-simd"
+	"sort"
+	"sync"
+)
+
+type innerPool interface {
+	Get(id types.TransactionId) (types.AddressableSignedTransaction, error)
+	GetAllItems() []types.AddressableSignedTransaction
+	Put(id types.TransactionId, item *types.AddressableSignedTransaction)
+	Invalidate(id types.TransactionId)
+}
+
+type TxPoolWithAccounts struct {
+	innerPool
+	accounts      map[types.Address]*pending_txs.AccountPendingTxs
+	accountsMutex sync.RWMutex
+}
+
+func NewTxPoolWithAccounts() *TxPoolWithAccounts {
+	return &TxPoolWithAccounts{
+		innerPool: NewTxMemPool(),
+		accounts:  make(map[types.Address]*pending_txs.AccountPendingTxs),
+	}
+}
+
+func (t *TxPoolWithAccounts) GetRandomTxs(numOfTxs int, seed []byte) []types.AddressableSignedTransaction {
+	txs := t.innerPool.GetAllItems()
+	if len(txs) <= numOfTxs {
```

Comment:
we should notice that this sort happens on each block creation and for each comparison between tx ids ,  GetTransactionId  hashes the transaction (currently transactionId is a calculated field) this will probably be expensive, 
a solution good solution IMO is for GetAllItems to return a map from txid to tx (the pool is aware of ids )

Reply:
This code changed - `id` is now cached, so it's a little less bad. If we have a performance issue we'll handle it in a later PR.